document/core/text/modules.rst

```diff
@@ -138,11 +130,11 @@ In that case, a :ref:`type index <syntax-typeidx>` is automatically inserted:
      \text{(}~\text{type}~~x~\text{)}~~\Tparam^\ast~~\Tresult^\ast \\
    \end{array}
 
-where :math:`x` is the smallest existing :ref:`type index <syntax-typeidx>` whose definition in the current module is the :ref:`function type <syntax-functype>` :math:`[t_1^\ast] \toF [t_2^\ast]`.
-If no such index exists, then a new :ref:`type definition <text-type>` of the form
+where :math:`x` is the smallest existing :ref:`type index <syntax-typeidx>` whose definition in the current module is the :ref:`final <syntax-final>` :ref:`function type <syntax-functype>` :math:`[t_1^\ast] \toF [t_2^\ast]`.
+If no such index exists, then a new :ref:`recursive type <text-rectype>` of the form
```

Comment:
Does this capture the constraint that only types in singleton rec groups are eligible to be found? IIRC, that's how we decided this should work.

Reply:
Ah, you are right, that's missing. Fixed and made more precise.